pkg/parser/additional/additional.go

```diff
@@ -0,0 +1,199 @@
+package additional
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/rs/zerolog/log"
+	"gopkg.in/yaml.v2"
+)
+
+func checkConvertingError(err error) {
+	if err != nil {
+		log.Error().Msg("Failed to convert 'string' into 'int'")
+	}
+}
+
+func getRegexMatch(regex, text string) string {
+	var re = regexp.MustCompile(regex)
+	clean := re.FindAllString(text, -1)
+
+	if clean == nil {
+		return ""
+	}
+
+	return clean[0]
+}
+
+func getExpirationDate(datetime string) [3]int {
+	// get notAfter datetime
+	expirationInfo := getRegexMatch(`(notAfter=)[a-zA-Z]+[ ]+[0-9]{1,2}[ ]([0-9]{2}:[0-9]{2}:[0-9]{2})[ ][0-9]{4}`, datetime)
+	clean := strings.Split(expirationInfo, "=")
+
+	months := map[string]int{
+		"Jan": 1,
+		"Feb": 2,
+		"Mar": 3,
+		"Apr": 4,
+		"May": 5,
+		"Jun": 6,
+		"Jul": 7,
+		"Aug": 8,
+		"Sep": 9,
+		"Oct": 10,
+		"Nov": 11,
+		"Dec": 12,
+	}
+
+	// pick month
+	cleanMonth := getRegexMatch(`^[a-zA-Z]+`, clean[1])
+	month := months[cleanMonth]
+
+	// pick year
+	cleanYear := getRegexMatch(`[0-9]{4}`, clean[1])
+	year, errYear := strconv.Atoi(cleanYear)
+	checkConvertingError(errYear)
+
+	// pick day
+	cleanDay := getRegexMatch(`[ ][0-9]{1,2}`, clean[1])
+	day, errDay := strconv.Atoi(strings.ReplaceAll(cleanDay, " ", ""))
+	checkConvertingError(errDay)
+
+	var date [3]int
+	date[0] = year
+	date[1] = month
+	date[2] = day
+
+	return date
+}
+
+func expirationDateFromPemCertificate(filePath string) [3]int {
+	out, err := exec.Command("openssl", "x509", "-in", filePath, "-noout", "-dates").Output()
+	if err != nil {
+		log.Error().Msgf("Failed to execute command 'openssl x509 -in %s -noout -dates'", filePath)
+	}
+
+	date := getExpirationDate(string(out))
+
+	return date
+}
+
+func getBitRSAKeyBit(filePath string) int {
+	out, err := exec.Command("openssl", "x509", "-in", filePath, "-text", "-noout").Output()
```

Comment:
This is going to add an external dependency to KICS, it will mandate the user to have `openssl` CLI installed which I don't think is a good idea right now. We would have to update our docs, add `openssl` to our docker images and etc...

Instead of using `openssl` to parse the PEM file have you seen these examples using go standard libraries to validate a certificate?

- https://github.com/timewasted/go-check-certs/blob/19b8ef135a9c83d7dd93e6a69c937fd04bdf7044/check-certs.go#L198-L205
- https://github.com/alanorth/check-tls-certs/blob/master/main.go

Reply:
Since, your intention is not to validate the certificate and just to extract the dates. Maybe this example is better:

First I extracted a x509 cert from google.com:

`true | openssl s_client -connect google.com:443 2>/dev/null | openssl x509 > google-ssl.pem`

Then:
```
package main

import (
	"crypto/x509"
	"encoding/pem"
	"fmt"
	"os"
)

func main() {
	certPEM, err := os.ReadFile("google-ssl.pem")
	if err != nil {
		panic("failed to find certificate")
	}

	block, _ := pem.Decode(certPEM)
	if block == nil {
		panic("failed to parse certificate PEM")
	}
	cert, err := x509.ParseCertificate(block.Bytes)
	if err != nil {
		panic("failed to parse certificate: " + err.Error())
	}

	fmt.Println(cert.NotAfter)
}
```

![image](https://user-images.githubusercontent.com/75368139/115960672-b16e8b80-a50a-11eb-8c3b-4810d3511e26.png)

In this way you won't even need to parse the abbreviation of the months (ex: "Jan", "Feb", etc..) just split the string

Reply:
Thank you so much, Rogério. I appreciate your suggestions. I already adopted them.